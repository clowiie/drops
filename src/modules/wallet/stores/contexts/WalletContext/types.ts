export interface ContextType {
  walletAddress: string
}
