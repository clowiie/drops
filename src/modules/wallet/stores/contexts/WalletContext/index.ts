import { createContext } from 'react'

import { ContextType } from './types'

const WalletContext = createContext<ContextType>({
  walletAddress: '',
})

export default WalletContext
