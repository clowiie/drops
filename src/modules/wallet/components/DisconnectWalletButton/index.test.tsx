import { render } from '@testing-library/react'

import createRouteWrapper from 'test/utils/createRouterWrapper'
import combineWrappers from 'test/utils/combineWrappers'
import createContextWrapper from 'test/utils/createContextWrapper'

import WalletContext from 'modules/wallet/stores/contexts/WalletContext'

import * as DisconnectWalletButtonType from '.'

describe('DisconnectWalletButton', () => {
  const { default: DisconnectWalletButton } =
    require('.') as typeof DisconnectWalletButtonType

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container } = render(<DisconnectWalletButton />, {
        wrapper: combineWrappers(
          createContextWrapper(WalletContext, {
            walletAddress: 'address',
          }),
          createRouteWrapper(
            {
              initialIndex: 0,
              initialEntries: ['/somepath'],
            },
            {},
          ),
        ),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(container.querySelector('a')?.href).toBe(
        'http://localhost/somepath?modalDisconnectWallet=true',
      )
    })

    it('should match snapshots with empty wallet address', () => {
      const { container } = render(<DisconnectWalletButton />, {
        wrapper: combineWrappers(
          createContextWrapper(WalletContext, {
            walletAddress: '',
          }),
          createRouteWrapper(
            {
              initialIndex: 0,
              initialEntries: ['/somepath'],
            },
            {},
          ),
        ),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(container.querySelector('a')?.href).toBe(
        'http://localhost/somepath?modalDisconnectWallet=true',
      )
    })
  })
})
