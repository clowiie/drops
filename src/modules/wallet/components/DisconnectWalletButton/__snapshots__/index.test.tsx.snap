// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisconnectWalletButton snapshots should match snapshots 1`] = `
<a
  class="bg-gray-700 hover:bg-gray-500 text-white p-2 rounded-lg text-sm"
  href="/somepath?modalDisconnectWallet=true"
>
  ad...ress
</a>
`;

exports[`DisconnectWalletButton snapshots should match snapshots with empty wallet address 1`] = `
<a
  class="bg-gray-700 hover:bg-gray-500 text-white p-2 rounded-lg text-sm"
  href="/somepath?modalDisconnectWallet=true"
/>
`;
